compiler/src/dotty/tools/dotc/core/GadtConstraint.scala

```diff
@@ -10,77 +10,25 @@ import util.{SimpleIdentitySet, SimpleIdentityMap}
 import collection.mutable
 import printing._
 
-import scala.annotation.internal.sharable
+object GadtConstraint:
+  def apply(): GadtConstraint = empty
```

Comment:
Should we have a "canonical" empty `GadtConstraint`, to avoid allocating an extra object every time we create a new constraint?